### ISSUE TITLE
Support partial/tilde/carret versions in `@version` syntax

### ISSUE THREAD

**Issue by Emilgardis** [taiki-e/install-action#25]
Would like to be able to specify the minimal (and max) version for a tool to install.

Would enable the following

```
uses: taiki-e/install-action@v1
      with:
        tool: cargo-llvm-cov@~0.5
```

**Comment by Emilgardis** [comment]
Encountered in https://github.com/cross-rs/cross/pull/1042#issuecomment-1264685777

**Comment by taiki-e** [comment]
Thanks for the suggestion. Omitting the patch/minor version is already supported in some packages, and what you actually need for your use case is not to support the (complex) semver requirement, but to support omitting the patch/minor version in other packages as well, right?

https://github.com/taiki-e/install-action/blob/e85ee51dd0ff4e1362366b809fb249d9a672f562/.github/workflows/ci.yml#L32-L33

**Comment by Emilgardis** [comment]
Yes! Nothing fancy needed

So this issue is only present on some of the packages, with some being those installed via `binstall` ?

**Comment by NobodyXu** [comment]
`cargo-binstall` already supports this, i.e. `cargo binstall 'cargo-llvm-cov@~0.5'` or `cargo binstall --version '~0.5' cargo-llvm-cov`.

**Comment by Emilgardis** [comment]
My comment was written in reverse, the issue is not present in those installed by `cargo-bininstall` I presume because of that

**Comment by NobodyXu** [comment]
> My comment was written in reverse, the issue is not present in those installed by `cargo-bininstall` I presume because of that

Oh sorry :D

**Comment by taiki-e** [comment]
It is the unexpected behavior that `~` is accepted when binstall is used as a fallback. If that was the intended behavior, officially supporting the package would always be a breaking change.

**Comment by NobodyXu** [comment]
> officially supporting a new package would always be a breaking change

Sorry that I didn't quite understand this.
Why would supporting a new package be a breaking change?

**Comment by taiki-e** [comment]
> > officially supporting a new package would always be a breaking change
> 
> Sorry that I didn't quite understand this. Why would supporting a new package be a breaking change?

1. Users install crate X using install-action's binstall fallback with `~` requirement. (install-action does not support crate X, but crate X support installation via binstall)
2. install-action now supports crate X. (No support for `~` requirements)
3. Workflow to install crate X using `~` requirement became broken because it `~` requirement is now not supported.

**Comment by NobodyXu** [comment]
> Workflow to install crate X using ~ requirement became broken because it ~ requirement is now not supported.

Thanks for the explanation, I got it.
So I think the solution is to reject version that contains any operator and make sure it only contains digits and `.`, I can submit a PR for this fix.

**Comment by NobodyXu** [comment]
> It is the unexpected behavior that `~` is accepted when binstall is used as a fallback. If that was the intended behavior, officially supporting a new package would always be a breaking change.

@taiki-e This is fixed in #26

**Comment by taiki-e** [comment]
This has been supported in 2.0.0 (`taiki-e/install-action@v2`).
